commercial/app/controllers/BookOffersController.scala

```diff
@@ -25,38 +24,32 @@ class BookOffersController(bookFinder: BookFinder, bestsellersAgent: Bestsellers
       (specificBooks ++ bestsellersAgent.bestsellersTargetedAt(segment)).distinctBy(_.isbn).take(4)
     }
 
+  private def isValidIsbn(isbn: String): Boolean = true
+
   def getBook = Action.async { implicit request =>
-    specificId map { isbn =>
-      bookFinder.findByIsbn(isbn) map {
-        _ map { book =>
-          Cached(10.minutes){
-            JsonComponent(Json.toJson(book))
-          }
-        } getOrElse {
-          Cached(componentMaxAge)(jsonFormat.nilResult)
+
+    lazy val failedLookupResponse: Result = Cached(30.seconds)(JsonNotFound())(request)
+    lazy val badRequestResponse: Future[Result] = Future.successful(Cached(1.day)(JsonComponent(JsNull))(request))
```

Comment:
Both have a similar name while being of different type. Could we make this more explicit?